superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/index.jsx

```diff
@@ -370,14 +372,18 @@ class HeaderActionsDropdown extends React.PureComponent {
             </Menu>
           )
         ) : null}
-        {editMode && (
-          <Menu.Item key={MENU_KEYS.SET_FILTER_MAPPING}>
-            <FilterScopeModal
-              className="m-r-5"
-              triggerNode={t('Set filter mapping')}
-            />
-          </Menu.Item>
-        )}
+        {editMode &&
+          !(
+            isFeatureEnabled(FeatureFlag.DASHBOARD_NATIVE_FILTERS) &&
+            isEmpty(dashboardInfo?.metadata?.filter_scopes)
```

Comment:
See my comment in the PR description. I think it's fine to look at the `filter_scopes` field (since this is what setting the filter mapping configures) rather than checking for the non-presence of filter-box charts.